src/app/Dashboard/AddCard.tsx

```diff
@@ -77,64 +89,121 @@ export const AddCard: React.FunctionComponent<AddCardProps> = (props: AddCardPro
     ];
   }, [DashboardCards]);
 
-  const handleToggle = React.useCallback(
-    (isOpen) => {
-      setSelectOpen(isOpen);
-    },
-    [setSelectOpen]
-  );
-
   const handleSelect = React.useCallback(
     (_, selection, isPlaceholder) => {
       setSelection(isPlaceholder ? '' : selection);
       setSelectOpen(false);
+
+      const c = {};
+      if (selection) {
+        for (const ctrl of getConfigByTitle(selection).propControls) {
+          c[ctrl.key] = ctrl.defaultValue;
+        }
+      }
+      setPropsConfig(c);
     },
-    [setSelection, setSelectOpen]
+    [setSelection, setSelectOpen, getConfigByTitle, setPropsConfig]
   );
 
   const handleAdd = React.useCallback(() => {
     setShowWizard(false);
-    dispatch(addCardIntent(getConfigByTitle(selection).component.name));
-  }, [dispatch, selection]);
+    const cardTitle = getConfigByTitle(selection).component.name;
+    dispatch(addCardIntent(cardTitle, propsConfig));
+  }, [setShowWizard, dispatch, addCardIntent, selection, propsConfig]);
 
   const handleStart = React.useCallback(() => {
     setShowWizard(true);
   }, [setShowWizard]);
 
   const handleStop = React.useCallback(() => {
+    setSelection('');
     setShowWizard(false);
-  }, [setShowWizard]);
+  }, [setSelection, setShowWizard]);
+
+  // custom nav for disabling subsequent steps (ex. configuration) if a card type hasn't been selected first
+  const customNav: CustomWizardNavFunction = React.useCallback(
+    (
+      isExpanded: boolean,
+      steps: WizardControlStep[],
+      activeStep: WizardControlStep,
+      goToStepByIndex: (index: number) => void
+    ) => {
+      return (
+        <WizardNav isExpanded={isExpanded}>
+          {steps
+            .filter((step) => !step.isHidden)
+            .map((step, idx) => (
+              <WizardNavItem
+                key={step.id}
+                id={step.id}
+                content={step.name}
+                isCurrent={activeStep.id === step.id}
+                isDisabled={step.isDisabled || (idx > 0 && !selection)}
+                stepIndex={step.index}
+                onNavItemClick={goToStepByIndex}
+              />
+            ))}
+        </WizardNav>
+      );
+    },
+    [selection]
+  );
 
   return (
     <>
       <Card isRounded isLarge>
         {showWizard ? (
-          <Wizard onClose={handleStop} onSave={handleAdd} height={300}>
+          <Wizard onClose={handleStop} onSave={handleAdd} height={640} nav={customNav}>
```

Comment:
It's 640px right? Should we have some more "adaptive" unit?

Reply:
It's px yes. I'll change it to use `rem`.